core/db/migrate/20160203152910_backfill_unreturned_exchange_orders.rb

```diff
@@ -0,0 +1,31 @@
+class BackfillUnreturnedExchangeOrders < ActiveRecord::Migration
+  disable_ddl_transaction!
+
+  def up
+    Spree::Order.complete.joins(:shipments).where('spree_orders.created_at > spree_shipments.created_at').find_each do |order|
+      return_items = Spree::ReturnItem.where(exchange_inventory_unit_id: order.inventory_unit_ids)
+      if return_items.any?
+        ra_ids = return_items.pluck(:return_authorization_id).compact
+        cr_ids = return_items.pluck(:customer_return_id).compact
+        ra_order_ids = Spree::ReturnAuthorization.where(id: ra_ids).pluck(:order_id)
+        cr_order_ids = Spree::CustomerReturn.where(id: cr_ids).map(&:order_id)
+
+        # Remove the unreturned exchange order id in case it has exchanges itself
+        order_ids = (ra_order_ids + cr_order_ids).compact.uniq - [order.id]
```

Comment:
Yeah, that's why I added the check to ensure the result is always only one order.